dace/codegen/targets/fpga.py

```diff
@@ -284,6 +285,22 @@ def fpga_ptr(name: str,
     return name
 
 
+def unqualify_fpga_array_name(sdfg: dace.SDFG, arr_name: str):
+    '''
+    Returns the unqualified array name if it refers to an array interface.
+    Otherwise return it as it is.
+    :param name: array name to unqualify
+    '''
+
+    unqualified = re.sub('_in$|_out$', '', arr_name)
+    unqualified = re.sub('^__', '', unqualified)
```

Comment:
we have a lot of __ in the beginning of things, like pystate. please remove

Reply:
Maybe I can double check that starts with `__` **and** ends with `_in`/`_out` to avoid incorrect substitutions